File: code.js
const ITERATIONS = 10000000;

function runBenchmark() {
  const start = performance.now();
  let sum = 0n; // Using BigInt to handle large numbers like in C

  // Perform the same computationally intensive task
  for (let i = 0; i < ITERATIONS; i++) {
    for (let j = 0; j < 100; j++) {
      sum += BigInt(i * j);
    }
  }

  const end = performance.now();
  const timeUsed = (end - start) / 1000; // Convert to seconds

  console.log(`js Result: ${sum}`);
  console.log(`js Time taken: ${timeUsed} seconds`);
}

// Run the benchmark
runBenchmark();

File: code.go
package main

import (
    "fmt"
    "time"
)

const ITERATIONS = 1000000

func main() {
    var sum int64 = 0
    start := time.Now()

    // Perform some computationally intensive task
    for i := 0; i < ITERATIONS; i++ {
        for j := 0; j < 100; j++ {
            sum += int64(i * j)
        }
    }

    duration := time.Since(start)

    fmt.Printf("Go Result: %d\n", sum)
    fmt.Printf("Go Time taken: %f seconds\n", duration.Seconds())
}
